File: pserver/tsch_python_code/beta_factor.py
#

import numpy as np
#from numpy import *


def beta_factor(max_win, link_cpdf, link_PRR):
    beta_factor = 0.0

    link_CW_meas = np.zeros((1, 2*max_win+1))
    link_CW_ind  = np.zeros((1, 2*max_win+1))

    for window in range(0, max_win):
        X0 = link_cpdf[max_win+1-window]
        X1 = link_cpdf[max_win+1+window]

        if X0 != 0:
            link_CW_meas[0][max_win + 1 - window] = X0
            link_CW_ind[0][max_win + 1 - window]  = link_PRR
            
        if X1 != 0:
            link_CW_meas[0][max_win + 1 + window] = (1.0 - X1)
            link_CW_ind[0][max_win + 1 + window]  = (1.0 - link_PRR)

    ############
    term1 = 0.0
    term2 = 0.0
    if any(link_CW_ind[0]):
        term1 = np.mean(link_CW_ind[0])

    if any(link_CW_meas[0]):
        term2 = np.mean(link_CW_meas[0])

    if term1 == 0 and term2 == 0:
        beta_factor = 0.0
    else:
        beta_factor = 1.0*(term1-term2)/term1

    return beta_factor

File: pserver/tsch_python_code/Code_Sept25_2011/DataProcessing/private/BetaFactor.m
function xx = BetaFactor(max_win, link_cpdf, link_PRR)


%% compute terms for CW distance
link_CW_meas = zeros(2*max_win+1,1);
link_CW_ind  = zeros(2*max_win+1,1);
for window = 1:max_win,
    x0 = link_cpdf(max_win+1-window);
    x1 = link_cpdf(max_win+1+window);
    if x0~=0,
        link_CW_meas(max_win+1-window) = x0; 
        link_CW_ind(max_win+1-window)  = link_PRR;
    end
    if x1~=0,
        link_CW_meas(max_win+1+window) = 1 - x1;
        link_CW_ind(max_win+1+window)  = 1 - link_PRR;
    end
end

if any(link_CW_ind),
    term1 = mean(link_CW_ind(link_CW_ind~=0));         % if link_CW_ind has at least a nonzero element, 
                                                       % the average is taken only on the positive elements                                                                                                                      
else
    term1 = 0;                                         % if all elements of link_CW_ind are zero, the average is zero
end
if any(link_CW_meas),
    term2 = mean(link_CW_meas(link_CW_meas~=0));       % if link_CW_meas has at least a nonzero element, 
                                                       % the average is taken only on the positive elements                                                                                                                      
else
    term2 = 0;                                         % if all elements of link_CW_meas are zero, the average is zero
end

if (term1 == 0) && (term2 == 0),
    beta_factor = 0;                                   % this can happen only when I have either all losses or all success
else 
    beta_factor  = (term1-term2)./term1;   
%         beta_factor  = (mean(link_CW_ind(find(link_CW_ind~=0)))-mean(link_CW_meas(find(link_CW_meas~=0))))./mean(link_CW_ind(find(link_CW_ind~=0)))
end

xx = beta_factor;
return;